tests/src/rules/no-amd.js

```diff
@@ -2,7 +2,7 @@ import { RuleTester } from 'eslint';
 import eslintPkg from 'eslint/package.json';
 import semver from 'semver';
 
-const ruleTester = new RuleTester();
+const ruleTester = new RuleTester({ parserOptions: { ecmaVersion: 2015, sourceType: 'module' } });
```

Comment:
why is this needed? did the default of RuleTester change?

Reply:
I've confirmed that previous tests seemed to rely on the scopeManager modified by the babel-eslint patch.

https://github.com/babel/babel-eslint/blob/v8.2.6/lib/patch-eslint-scope.js#L363

If we do not specify `sourceType:'module'` in a normal scopeManager, this rule will not always return an error because `scope.type ='module'` will not be created.

This change was necessary as the patch would fail when using eslint v8.

Reply:
So, to be clear, this change is both correct and unnecessary in master? perhaps i could peel it off as a separate commit and land it first.

Reply:
If you run the following command in the repository before making this PR change, the test will fail.

`npx cross-env BABEL_ENV=test nyc -s mocha tests/src/rules/no-amd.js `

I will create another PR to fix this.

Reply:
I opened PR #2193.